cli/cmd/stat_summary.go

```diff
@@ -0,0 +1,254 @@
+
+func init() {
+	RootCmd.AddCommand(statSummaryCommand)
+	addControlPlaneNetworkingArgs(statSummaryCommand)
+	// TODO: the -n flag is taken up by conduit-namespace :( we should move it to something else so this can have -n
+	statSummaryCommand.PersistentFlags().StringVarP(&namespace, "namespace", "a", "default", "namespace of the specified resource")
+	statSummaryCommand.PersistentFlags().StringVarP(&timeWindow, "time-window", "t", "1m", "Stat window.  One of: '10s', '1m', '10m', '1h'.")
+
+	statSummaryCommand.PersistentFlags().StringVarP(&outToName, "out-to", "", "", "If present, restricts outbound stats to the specified resource name")
+	statSummaryCommand.PersistentFlags().StringVarP(&outToNamespace, "out-to-namespace", "", "", "Sets the namespace used to lookup the '--out-to' resource. By default the current '--namespace' is used")
+	statSummaryCommand.PersistentFlags().StringVarP(&outToType, "out-to-resource", "", "", "If present, restricts outbound stats to the specified resource type")
+
+	statSummaryCommand.PersistentFlags().StringVarP(&outFromName, "out-from", "", "", "If present, restricts outbound stats to the specified resource name")
+	statSummaryCommand.PersistentFlags().StringVarP(&outFromNamespace, "out-from-namespace", "", "", "Sets the namespace used to lookup the '--out-to' resource. By default the current '--namespace' is used")
+	statSummaryCommand.PersistentFlags().StringVarP(&outFromType, "out-from-resource", "", "", "If present, restricts outbound stats to the specified resource type")
+}
+
+func requestStatSummaryFromAPI(client pb.ApiClient) (string, error) {
+	req, err := buildStatSummaryRequest()
+
+	if err != nil {
+		return "", fmt.Errorf("error creating metrics request while making stats request: %v", err)
+	}
+
+	resp, err := client.StatSummary(context.Background(), req)
+	if err != nil {
+		return "", fmt.Errorf("error calling stat with request: %v", err)
+	}
+
+	return renderStatSummary(resp)
+}
+
+func renderStatSummary(resp *pb.StatSummaryResponse) (string, error) {
+	var buffer bytes.Buffer
+	w := tabwriter.NewWriter(&buffer, 0, 0, padding, ' ', tabwriter.AlignRight)
+
+	writeStatTableToBuffer(resp, w)
+	w.Flush()
+
+	// strip left padding on the first column
+	out := string(buffer.Bytes()[padding:])
+	out = strings.Replace(out, "\n"+strings.Repeat(" ", padding), "\n", -1)
+
+	return out, nil
+}
+
+type summaryRow struct {
+	meshed      string
+	requestRate float64
+	successRate float64
+	latencyP50  int64
+	latencyP99  int64
+}
+
+func writeStatTableToBuffer(resp *pb.StatSummaryResponse, w *tabwriter.Writer) {
+	nameHeader := "NAME"
+	maxNameLength := len(nameHeader)
+
+	stats := make(map[string]*summaryRow)
+
+	for _, statTable := range resp.GetOk().StatTables {
+		table := statTable.GetPodGroup()
+		for _, r := range table.Rows {
+			name := r.Resource.Name
+			if name == "" {
+				continue
+			}
+
+			if len(name) > maxNameLength {
+				maxNameLength = len(name)
+			}
+
+			if _, ok := stats[name]; !ok {
```

Comment:
hmmmm.... I don't think this should happen, since we create an empty map at the beginning of this function, and a duplicate entry would imply there were two rows with the same resource name, which shouldn't happen.